### ISSUE TITLE
Incorrect intellisense error msg in rest params

### ISSUE THREAD

**Issue by Treri** [Microsoft/TypeScript#15428]
**TypeScript Version:**  vscode 1.11.2 builtin typescript

**Code**

```ts
interface buttonObject {
  text: string;
  title: string;
  color?: boolean
}

interface Func {
  (...groups: Array<buttonObject | Array<buttonObject>>): any
}

let func: Func = function(...groups){
  let result = groups.map(function(buttons){
    if(Object.prototype.toString.call(buttons) === '[object Object]'){
      return [buttons];
    }else{
      return buttons;
    }
  });

  // do something else
}
```

Then I invoke func, if I passed a object, the intellisense error msg is incorrect

```ts
func({
  text: 123,
  title: 'title'
})
```

**Expected behavior:**
```
[ts]
Argument of type '{ text: number; title: string; }' is not assignable to parameter of type 'buttonObject'.
  Types of property 'text' are incompatible.
    Type 'number' is not assignable to type 'string'.
```

**Actual behavior:**
```
[ts]
Argument of type '{ text: number; title: string; }' is not assignable to parameter of type 'buttonObject | buttonObject[]'.
  Type '{ text: number; title: string; }' is not assignable to type 'buttonObject[]'.
    Property 'length' is missing in type '{ text: number; title: string; }'.
```

**AND**
If I import the `buttonObject` interface from this file outside and   
If I passed a object to `func`, there is no intellisense when I write object property. But if I put the object in an array, the intellisense worked again.

```ts
func({ // have no intellisense
  text: 123,
  title: 'title'
})

func([{ // intellisense works
  text: 123,
  title: 'title'
}])
```

screenshot

![image](https://cloud.githubusercontent.com/assets/2631733/25512651/e52cd03e-2c01-11e7-8fde-b062e6773f1b.png)

**Comment by mhegazy** [comment]
The error message is correct since your type is `buttonObject | buttonObject[]`, the error is reported only on one of the constituents of the union type, it just happened to be the array one..

Intellisense not working, that is a bug. it is tracked by https://github.com/Microsoft/TypeScript/issues/6405

**Comment by andy-ms** [comment]
Fixed by #15541
